server/routes/sockets/room-service.js

```diff
@@ -147,23 +199,35 @@ class RoomService {
     /**
      * Have the user close the specified room.
      */
-    clientCloseRoomEvent(eventInfo, socket) {
-        if (!eventInfo.roomID) {
+    async clientCloseRoomEvent(eventInfo, socket) {
+        if (
+            !eventInfo.roomID
+        ) {
             Utils.DebugLog('Invalid event info passed to clientCloseRoomEvent.');
             return;
         }
         try {
+            // Grab the correct room from the list of active rooms
+            const room = this.activeRoomsByID[eventInfo.roomID];
             // If the specified roomID is not in the list of active rooms, we cannot close the room. Emit an error event.
-            if (!this.activeRoomsByID[eventInfo.roomID]) {
+            if (!room) {
                 this.emitUserEvent('host_room_closed_failure', socket);
-            } else {
-                // Grab the correct room from the list of active rooms
-                const room = this.activeRoomsByID[eventInfo.roomID];
-                // Close the room
-                this.closeRoom(room);
-                // Emit a success message
-                this.emitUserEvent('host_room_closed_success', socket);
+                return;
+            }
+            // Check if the user is authorized as the host of a room.
+            if (!await this.checkIfUserAuthorizedForRoom(
+                room, eventInfo.authToken, socket,
+                {
+                    "title": "Failed to Close Room",
+                    "message": "We could not authorize your attempt to close this room."
+                }
+            )) {
+                return;
```

Comment:
Same feedback as above comment, and ditto for the remaining places this pattern appears